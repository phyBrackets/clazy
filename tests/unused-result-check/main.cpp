#include <QtCore/QObject>

class A : public QObject {
  Q_OBJECT
public:
  int g;
  A() : g(foo()) {} // OK
  int foo() const { return 5; }
  void bar() const {
    foo();     // WARN
    if (foo()) // OK
    {
    }
  }
};

class B {
public:
  A a;
  B() : g(a.foo()) {} // OK
  int g;
};

int main(int argc, char *argv[]) {
  A a;
  B b;
  a.bar();
  auto lambda = [](A &obj) { obj.foo(); }; // WARN
  lambda(a);
  return 0;
}
