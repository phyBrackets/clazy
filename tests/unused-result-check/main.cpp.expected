unused-result-check/main.cpp:10:5: warning: Result of const member function is not used. [-Wclazy-unused-result-check] 
unused-result-check/main.cpp:28:34: warning: Result of const member function is not used. [-Wclazy-unused-result-check]
